src/reanimated2/fabricUtils.ts

```diff
@@ -1,33 +1,65 @@
 'use strict';
 /* eslint-disable camelcase */
 
-import { isFabric } from './PlatformChecker';
 import type { ShadowNodeWrapper } from './commonTypes';
 
-interface HostInstance {
-  _internalInstanceHandle: {
-    stateNode: {
-      node: ShadowNodeWrapper;
-    };
-  };
-}
-
-let findHostInstance_DEPRECATED: (ref: React.Component) => HostInstance;
+let findHostInstance_DEPRECATED: (ref: unknown) => void;
+let getInternalInstanceHandleFromPublicInstance: (ref: unknown) => {
+  stateNode: { node: unknown };
+};
 
 export function getShadowNodeWrapperFromRef(
```

Comment:
Also, can we refactor this file a bit? because is really hard to read

Reply:
Since we are in a bit of hurry with bridgeless mode, can we merge it as it is as long as it works? Then we can make a follow-up for it to properly handle it with some links to code from where I took those shenanigans.

Reply:
Me and @piaskowyk had the same idea here.